### PR TITLE
Clickable icon config

### PR DESCRIPTION
**Pull request by MartinRiese**
## Product Description

Add a new format in case lists to specify clickable icons. In addition to the normal icon options add drop down menu to
select from auto submitting forms.

image

## Technical Summary

https://dimagi-dev.atlassian.net/browse/USH-3529

## Feature Flag

CASE_LIST_CLICKABLE_ICON

### Safety story

Behind a feature flag

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/details/utils.js [hunk 1]
```diff
@@ -45,8 +45,15 @@ hqDefine("app_manager/js/details/utils", function () {
...
-            label: gettext('Address Popup (Web Apps only)'),
+                value: "address-popup",
+                label: gettext('Address Popup (Web Apps only)'),
+            });
+        }
+
+        if (hqImport('hqwebapp/js/toggles').toggleEnabled('CASE_LIST_CLICKABLE_ICON')) {
+            formats.push({
+                value: "clickable-icon",
+                label: gettext('Clickable Icon (Web Apps only)'),
```
Comment (snopoke) [inline corehq/apps/app_manager/static/app_manager/js/details/utils.js:56]: Just wondering about this name. Might be good to check with product what to call it in the UI.

File: corehq/apps/hqwebapp/tests/data/bootstrap5_diffs/javascript/hqwebapp/ui_elements/ui-element-select.js.diff.txt [hunk 2]
```diff
@@ -10,3 +10,14 @@
...
+         setOptions: function (options) {
+             this.options = options.map(o => _.isString(o) ? {value: o, label: o} : o);
+             this.$edit_view.html('');
+-            for (let i = 0; i < this.options.length; i += 1) {
+-                const option = this.options[i];
++            for (var i = 0; i < this.options.length; i += 1) {
++                var option = this.options[i];
+                 if ('groupName' in option) {
+                     $(`<optgroup label="${option.groupName}"/>`).appendTo(this.$edit_view);
+                 } else {
```
Comment (snopoke) [inline corehq/apps/hqwebapp/tests/data/bootstrap5_diffs/javascript/hqwebapp/ui_elements/ui-element-select.js.diff.txt:23]: I think this diff indicates that you should make the same changes to the 'bootstrap3' version of the `ui-element-select.js` file.

**Review by snopoke — COMMENTED** [review]
Looks great.

File: corehq/apps/app_manager/static/app_manager/js/details/utils.js (hunk 1 above)
Comment (MartinRiese) [reply corehq/apps/app_manager/static/app_manager/js/details/utils.js:56]: I see you changed that in your PR. So I think we are good here.

File: corehq/apps/hqwebapp/tests/data/bootstrap5_diffs/javascript/hqwebapp/ui_elements/ui-element-select.js.diff.txt (hunk 2 above)
Comment (MartinRiese) [reply corehq/apps/hqwebapp/tests/data/bootstrap5_diffs/javascript/hqwebapp/ui_elements/ui-element-select.js.diff.txt:23]: Yes, I made the changes to both bootstrap 3 and 5

File: corehq/apps/app_manager/static/app_manager/js/details/utils.js (hunk 1 above)
Comment (snopoke) [reply corehq/apps/app_manager/static/app_manager/js/details/utils.js:56]: Did I? I don't think I did.

**Review by snopoke — APPROVED** [review]

File: corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/bootstrap5/ui-element-select.js [hunk 3]
```diff
@@ -84,7 +82,11 @@ hqDefine('hqwebapp/js/ui_elements/bootstrap5/ui-element-select', [
             this.$edit_view.html('');
             for (var i = 0; i < this.options.length; i += 1) {
                 var option = this.options[i];
-                $('<option/>').text(option.label).val(option.value).appendTo(this.$edit_view);
+                if ('groupName' in option) {
```
Comment (biyeun) [inline corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/bootstrap5/ui-element-select.js:85]: thank you for updating both files. the changes look good ✅

**Review by snopoke — APPROVED** [review]

**Pull request merged**